json/concatenated_json_parse_stream.ts

```diff
@@ -12,32 +12,90 @@ const primitives = new Map(
 );
 
 /**
- * Stream to parse {@link https://en.wikipedia.org/wiki/JSON_streaming#Concatenated_JSON|Concatenated JSON}.
+ * Stream to parse
+ * {@link https://en.wikipedia.org/wiki/JSON_streaming#Concatenated_JSON | Concatenated JSON}.
+ *
+ * @example Usage
  *
- * @example
  * ```ts
  * import { ConcatenatedJsonParseStream } from "@std/json/concatenated-json-parse-stream";
+ * import { assertEquals } from "@std/assert/assert-equals";
  *
- * const url = "@std/json/testdata/test.concatenated-json";
- * const { body } = await fetch(url);
- *
- * const readable = body!
- *   .pipeThrough(new TextDecoderStream()) // convert Uint8Array to string
- *   .pipeThrough(new ConcatenatedJsonParseStream()); // parse Concatenated JSON
+ * const stream = ReadableStream.from([
+ *   `{"foo":"bar"}`,
+ *   `{"baz":100}`,
+ * ]).pipeThrough(new ConcatenatedJsonParseStream());
  *
- * for await (const data of readable) {
- *   console.log(data);
- * }
+ * assertEquals(await Array.fromAsync(stream), [
+ *   { foo: "bar" },
+ *   { baz: 100 },
+ * ]);
  * ```
  */
 export class ConcatenatedJsonParseStream
   implements TransformStream<string, JsonValue> {
-  /** A writable stream of byte data. */
+  // TODO(iuioiua): Investigate why this class is implemented differently to the other JSON streams.
```

Comment:
`JSONParseStream` supposes every chunk is valid JSON string which represents a single JSON object. `ConcatenatedJsonParseStream` handles stream of strings where each chunk can be incomplete fragments of JSON strings.

This difference exists because in NDJSON or JSONLines, each line contains single JSON object. So we can use `TextLineStream` to split them and pass the result to `JSONParseStream`. On the other hand, in Concatenated JSON, there's no explicit delimiter of JSONs. So the class needs to maintain the internal parsing state of JSONs